101-hdinsight-linux-ssh-publickey/azuredeploy.json

```diff
@@ -99,10 +94,10 @@
         "storageProfile": {
           "storageaccounts": [
             {
-              "name": "[concat(parameters('clusterStorageAccountName'),'.blob.core.windows.net')]",
+              "name": "[replace(replace(concat(reference(concat('Microsoft.Storage/storageAccounts/', variables('storageAccountName')), variables('defaultApiVersion')).primaryEndpoints.blob),'https:',''),'/','')]",
```

Comment:
@philon-msft Thanks :) And I totally agree. I'm fixing all the templates with dynamic endpoints, replacing the hardcoded values. Its an huge task, having a consistent replacement makes it achievable.